orttraining/orttraining/training_ops/cuda/communication/recv.cc

```diff
@@ -66,6 +74,20 @@ Status Recv::ComputeInternal(OpKernelContext* ctx) const {
     info_shape_sizes.buffer, info_shape_sizes.size, MPI_CHAR,
     info_shape_sizes.rank, info_shape_sizes.tag, MPI_COMM_WORLD, MPI_STATUS_IGNORE);
   ORT_ENFORCE(mpi_code == MPI_SUCCESS, "MPI Recv fails.");
+
+#ifndef NDEBUG
+  // This range object includes the first MPI_Recv which receives a scalar.  
+  // It means we count the MPI's initialization in pre-recv stage.
+  preRange.End();
+#endif
+
+#ifndef NDEBUG
+  profile::NvtxRangeCreator recvRange(
+    "Recv-" + std::to_string(src), profile::Green);
```

Comment:
>std::to_string(src) [](start = 14, length = 19)

the MakeString() function may be useful (common.h)

Reply:
I would like to make this library as independent as possible. This won't hurt because we only do profiling for Debug build.

---
In reply to: [428992562](https://github.com/microsoft/onnxruntime/pull/4013#discussion_r428992562) [](ancestors = 428992562)

Reply:
I was referring to something like `MakeString("Recv-", src)` here. The library in the profile namespace doesn't depend on it.